### PR TITLE
add some missing child_attrs

### PR DESCRIPTION
**Pull request by MarcoGorelli**
adding in some missing child_attrs as per https://github.com/cython/cython/issues/1628#issuecomment-1416517185

currently I'm having to hack around them in cython-lint:

https://github.com/MarcoGorelli/cython-lint/blob/2ecb4db23cabbfd63cc825a3bca638ae42e9434b/cython_lint/cython_lint.py#L777-L801

### CODE REVIEW

File: Cython/Compiler/ExprNodes.py [hunk 1]
```diff
@@ -7216,6 +7217,7 @@ class AttributeNode(ExprNode):
     #  is_called            boolean   Function call is being done on result
     #  entry                Entry     Symbol table entry of attribute
 
+    child_attrs = ["attribute"]
```
Comment (da-woods) [inline Cython/Compiler/ExprNodes.py:7220]: This definitely doesn't fit our `child_attrs` scheme - `attribute` is a string, but the `child_attrs` should inherit from `Node` (or be lists of things that inherit from `Node`)

File: Cython/Compiler/ExprNodes.py [hunk 2]
```diff
@@ -14401,6 +14405,7 @@ class AnnotationNode(ExprNode):
     # Doesn't handle the pre PEP-563 version where the
     # annotation is evaluated into a Python Object.
 
+    child_attrs = ["expr"]
```
Comment (da-woods) [inline Cython/Compiler/ExprNodes.py:14408]: This forces `expr` to be processed correctly in Cython, while the point PEP-563 is that it shouldn't have to be. So may cause problems. (We may revise this though)

File: Cython/Compiler/Nodes.py [hunk 3]
```diff
@@ -7505,6 +7505,7 @@ def annotate(self, code):
 class ForInStatNode(_ForInStatNode):
     #  'for' statement
 
+    child_attrs = ["target"]
```
Comment (da-woods) [inline Cython/Compiler/Nodes.py:7508]: Is this not covered by the base class `_ForInStateNode`?

**Review by da-woods — COMMENTED** [review]
I wonder if we could provide an alternative mechanism for you (and other similar tools) - `unvisited_child_attrs` or similar. To mark things that are deliberately ignored by Cython, but which third-party tools might care about.

I wouldn't be oppose to that as a solution.

**Comment by MarcoGorelli** [comment]
> I wonder if we could provide an alternative mechanism for you (and other similar tools) - unvisited_child_attrs or similar. To mark things that are deliberately ignored by Cython, but which third-party tools might care about.
> I wouldn't be oppose to that as a solution.

If possible, that would be awesome, thanks! Closing for now then

**Pull request closed without merging**